src/codegen/mod.rs

```diff
@@ -237,6 +239,23 @@ fn contract(contract_no: usize, ns: &mut Namespace, opt: &Options) {
             ns.contracts[contract_no].default_constructor = Some((func, cfg_no));
         }
 
+        // TODO: This is a temporary solution. Once Substrate's dispatch moves to codegen,
```

Comment:
I noticed you implemented new mod `dispatch.rs` together with some changes already in the substrate emit already; what would be left to make this `TODO` happen? And what do you think how pressing would this be for me to take care of?

Reply:
The old `emit/dispatch.rs` has been moved to `emit/substrate/dispatch.rs` because it is now Substrate specific. The changes I made in the file were only supposed to remove Solana specific code that is not needed anymore. 

If you have a look at the new `codegen/dispatch.rs`, you'll see that the file is very similar to `emit/substrate/dispatch.rs` with few exceptions that are substrate specific. You'll need then to implement these details and build a call to the new dispatch functions wherever `emit_function_dispatch` function is called. The idea I had for Substrate was to just reuse the Solana code in codegen and build a function call in LLVM-IR to the dispatch we generate in codegen. This is the easiest way, but not the prettiest one in my opinion.

Reply:
Alright, that sounds like a plan. I'll look into this later on when I find some time. Thanks! :heart: